.atomist/editors/NewSpringBootRestService.ts

```diff
@@ -0,0 +1,170 @@
+        validInput: "a valid Maven artifact ID, which starts with a lower-case letter and contains only alphanumeric, -, and _ characters",
+        minLength: 1,
+        maxLength: 21,
+        required: false
+    })
+    artifact_id: string = "myartifact";
+
+    @Parameter({
+        displayName: "Group ID",
+        description: "Maven group identifier, often used to provide a namespace for your project, e.g., com.pany.team",
+        pattern: "^[a-z][-a-z0-9_]*$", // Ideally this should be looking up artifact_id as a common pattern
+        validInput: "a valid Maven group ID, which starts with a letter, -, or _ and contains only alphanumeric, -, and _ characters and may having leading period separated identifiers starting with letters or underscores and containing only alphanumeric and _ characters.",
+        minLength: 1,
+        maxLength: 21,
+        required: false
+    })
+    group_id: string = "mygroup";
+
+    @Parameter({
+        displayName: "Version",
+        description: "initial version of the project, e.g., 1.2.3-SNAPSHOT",
+        pattern: Pattern.semantic_version,
+        validInput: "a valid semantic version, http://semver.org",
+        minLength: 1,
+        maxLength: 21,
+        required: false
+    })
+    version: string = "0.1.0-SNAPSHOT";
+
+    @Parameter({
+        displayName: "Project Description",
+        description: "short descriptive text describing the new project",
+        pattern: Pattern.any,
+        validInput: "free text",
+        minLength: 1,
+        maxLength: 50,
+        required: false
+    })
+    description: string = "My new project"
+
+    @Parameter({
+        displayName: "Root Package",
+        description: "root package for your generated source, often this will be namespaced under the group ID",
+        pattern: Pattern.java_package,
+        validInput: "a valid Java package name, which consists of period-separated identifiers which have only alphanumeric characters, $ and _ and do not start with a number",
+        minLength: 1,
+        maxLength: 50,
+        required: false
+    })
+    root_package: string = "com.myorg";
+
+    @Parameter({
+        displayName: "Class Name",
+        description: "name for the service class",
+        pattern: Pattern.java_class,
+        validInput: "a valid Java class name, which contains only alphanumeric characters, $ and _ and does not start with a number",
+        minLength: 1,
+        maxLength: 50,
+        required: false
+    })
+    service_class_name: string = "com.myorg";
+
+    populate(project: Project) {
+
+        this.switchREADME(project);
+
+        this.pomParameterizer(project);
+
+        this.packageMove(project, this.root_package);
+
+        this.renameServiceClass(project, this.service_class_name)
+
+        this.addCreationDateToReadme(project)
+
+        this.tidyProject(project);
+    }
+
+    switchREADME(project) {
```

Comment:
Shouldn't all of these function signatures provide a type for `project`, i.e., `project: Project`?